### PR TITLE
Refactor usages of `onElementCreated` and `onElementUpdated`

### PR DESCRIPTION
**Pull request by agubler**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

**Description:**

Refactored usages of `onElementCreated` and `onElementUpdated`in widgets as the hooks are being removed from [`@dojo/widget-core/WidgetBase`](https://github.com/dojo/widget-core/pull/840).

### CODE REVIEW

File: src/dialog/Dialog.ts [hunk 1]
```diff
@@ -81,11 +82,9 @@ export class DialogBase<P extends DialogProperties = DialogProperties> extends T
...
 	}
 
-	protected onElementCreated(element: HTMLElement, key: string) {
-		if (key === 'main') {
-			const { open = false } = this.properties;
-			open && element.focus();
-		}
+	protected onAttach() {
+		const { open = false } = this.properties;
+		open && this.meta(Focus).set('main');
```
Comment (smhigley) [inline src/dialog/Dialog.ts]: this isn't working, `onAttach` is only called when the widget as a whole is added to the DOM, not when the dialog element is created and attached.

Comment (smhigley) [reply src/dialog/Dialog.ts]: I forgot to add functional focus tests when I added focus handling to the dialog, but those would've been useful 😝

File: src/splitpane/SplitPane.ts [hunk 2]
```diff
@@ -39,11 +40,11 @@ const DEFAULT_SIZE = 100;
 
 @theme(css)
 export class SplitPaneBase<P extends SplitPaneProperties = SplitPaneProperties> extends ThemedBase<P, null> {
-	private _divider: HTMLElement;
+	// private _divider: HTMLElement;
 	private _dragging: boolean;
 	private _lastSize?: number;
 	private _position: number;
-	private _root: HTMLElement;
+	// private _root: HTMLElement;
```
Comment (smhigley) [inline src/splitpane/SplitPane.ts]: Can we remove these?

File: src/dialog/Dialog.ts (hunk 1 above)
Comment (agubler) [reply src/dialog/Dialog.ts]: Ha that would have let me know it wasn't working!

File: src/select/Select.ts [hunk 3]
```diff
@@ -159,14 +159,6 @@ export class SelectBase<P extends SelectProperties = SelectProperties> extends T
...
 	}
 
-	protected onElementUpdated(element: HTMLElement, key: string) {
-		if (key === 'root' && this._callListboxFocus) {
-			this._callListboxFocus = false;
-			const listbox = <HTMLElement> element.querySelector('[role="listbox"]');
-			listbox && listbox.focus();
-		}
-	}
-
```
[the remark this answers is not in the document]
Comment (agubler) [reply src/select/Select.ts]: Think we're going to have to raise an issue to address this retrospectively (again no functional test failures)

[the remark this answers is not in the document]
Comment (smhigley) [reply src/select/Select.ts]: It probably needs the same treatment as Combobox/TextInput, where Listbox would need a `focus` property

[the remark this answers is not in the document]
Comment (smhigley) [reply src/select/Select.ts]: I'm hesitant to merge the PR with broken keyboard functionality for Select, though

[the remark this answers is not in the document]
Comment (agubler) [reply src/select/Select.ts]: Should be sending focus in the listbox now.

File: src/dialog/Dialog.ts (hunk 1 above)
Comment (agubler) [reply src/dialog/Dialog.ts]: Have moved this to be called when the `open` property is `true` and the dialog was not open previously (already had this state in the widget)

**Pull request merged**